### PR TITLE
Update

### PR DESCRIPTION
**Pull request by versokova**
info in commit messages

### CODE REVIEW

File: sl/symheap.hh [hunk 1]
```diff
@@ -560,6 +560,9 @@ class SymHeapCore {
         /// clear the list of anonymous stack objects of the given call instance
         void clearAnonStackObjects(TObjList &dst, const CallInst &of);
 
+        /// remove anonymous stack object
+        void removeAnonStackObj(const TObjId obj, const CallInst &from);
```
Comment (kdudka) [inline sl/symheap.hh:564]: The current interface of `SymHeap` for handling anonymous stack objects is ugly.  This change does not make it any better.  I propose to replace the `clearAnonStackObjects()` method by something that just returns the list of objects.  The list of anonymous stack objects in `SymHeap` should be updated automatically each time `objDestroy()` is called.

Let me create a separate pull request to get this fixed first.  Then I will rebase this pull request on top of it.

Is there an easy way to check that this change still works as expected?

It does not seem to be covered by existing regression tests.  I tried to run them with `gcc-9.3.0` and `llvm-9.0.1` with this patch applied and nothing seemed to be broken at first glance:
```
--- a/sl/symproc.cc
+++ b/sl/symproc.cc
@@ -2849,19 +2849,19 @@ void SymExecCore::handleClobber(const CodeStorage::Insn &insn)
             this->setLocation(varLoc);

         CL_DEBUG_MSG(lw_, "FFF SymExecCore::handleClobber() destroys var "
                 << varString);
     }
-
+#if 0
     // destroy the target and collect the junk
     if (sh_.isAnonStackObj(obj)) {
         // if variable was allocated on stack via __builtin_alloca
         // need to be remove from AnonStackObjects
         const CallInst callInst(this->bt_);
         sh_.removeAnonStackObj(obj, callInst);
     }
-
+#endif
     CL_DEBUG_MSG(lw_, "FFF SymExecCore::handleClobber() destroys object #"
                 << obj);

     this->objDestroy(obj);
 }
```

Comment (versokova) [reply sl/symheap.hh:564]: I wanted to debug Predator to find unimplemented things and almost all regression tests (with LLVM/debug mode and clobber instruction) failed here.
`objDestroy` is called two times over the same object, first from `handleClobber` and then from `execStackRestore`.
```
nika$ clang -w -g -m32 -S -emit-llvm -Xclang -fsanitize-address-use-after-scope -Iinclude/predator-builtins -DPREDATOR tests/predator-regre/test-0154.c -o - |  opt -o /dev/null -lowerswitch -load passes-src/passes_build/libpasses.dylib -nestedgep -load sl_build/libsl.dylib -sl
tests/predator-regre/test-0154.c:143:5: note: writing heap graph to 'test-0154.c-143-0000.dot'... [-sl]
/Users/nika/Documents/REPOS/repo-predator-sym/sl/symgc.cc:143: conditional breakpoint fired: CL_BREAK_IF(!sh.isValid(obj))
/Users/nika/Documents/REPOS/repo-predator-sym/sl/symgc.cc:143: killing self by INT3 [SHA1 9961a7f2efe1549d2e476164f510df6c3844c1bc]
Stack dump:
0.	Program arguments: /usr/local/Cellar/llvm/10.0.0_1/bin/opt -o /dev/null -lowerswitch -load passes-src/passes_build/libpasses.dylib -nestedgep -load sl_build/libsl.dylib -sl
0  libLLVM.dylib            0x00000001083f673e llvm::sys::PrintStackTrace(llvm::raw_ostream&) + 37
1  libLLVM.dylib            0x00000001083f6ba0 SignalHandler(int) + 220
2  libsystem_platform.dylib 0x00007fff55fa5f5a _sigtramp + 26
3  libsystem_platform.dylib 0x000000000000ffff _sigtramp + 18446603339073757375
4  libsl.dylib              0x000000010cec4caf LeakMonitor::destroyObject(TObjId) + 47
5  libsl.dylib              0x000000010cec4bdf SymProc::objDestroy(TObjId) + 159
6  libsl.dylib              0x000000010cecddf9 SymExecCore::execStackRestore() + 2169
7  libsl.dylib              0x000000010cdd0791 SymExecEngine::execReturn() + 1505
8  libsl.dylib              0x000000010cdd6545 SymExecEngine::execTermInsn() + 101
9  libsl.dylib              0x000000010cdd77ef SymExecEngine::execInsn() + 3903
10 libsl.dylib              0x000000010cdd88f2 SymExecEngine::execBlock() + 2050
11 libsl.dylib              0x000000010cddaf26 SymExecEngine::run() + 2262
12 libsl.dylib              0x000000010cde0f0c SymExec::execFnc(SymState&, SymHeap const&, CodeStorage::Insn const&, CodeStorage::Fnc const&) + 636
13 libsl.dylib              0x000000010cde1f8a execTopCall(SymState&, SymHeap const&, CodeStorage::Insn const&, CodeStorage::Fnc const&) + 138
14 libsl.dylib              0x000000010cde328f execute(SymState&, SymHeap const&, CodeStorage::Fnc const&) + 1871
15 libsl.dylib              0x000000010cc90e47 execFnc(CodeStorage::Fnc const&, bool) + 1591
16 libsl.dylib              0x000000010cc93f3b launchSymExec(CodeStorage::Storage const&) + 5419
17 libsl.dylib              0x000000010cc94096 clEasyRun(CodeStorage::Storage const&, char const*) + 278
18 libsl.dylib              0x000000010cb0e494 ClEasy::run(CodeStorage::Storage&) + 9476
19 libsl.dylib              0x000000010cb311a0 ClStorageBuilder::acknowledge() + 32
20 libsl.dylib              0x000000010cb6951a ClFilterBase::acknowledge() + 26
21 libsl.dylib              0x000000010cb69077 ClfCbSeqChk::acknowledge() + 39
22 libsl.dylib              0x000000010cb6951a ClFilterBase::acknowledge() + 26
23 libsl.dylib              0x000000010cb6951a ClFilterBase::acknowledge() + 26
24 libsl.dylib              0x000000010cb98006 cl_wrap_acknowledge(cl_code_listener*) + 54
25 libsl.dylib              0x000000010cae5c1d ClChain::acknowledge() + 333
26 libsl.dylib              0x000000010cb98006 cl_wrap_acknowledge(cl_code_listener*) + 54
27 libsl.dylib              0x000000010caa7554 CLPass::doFinalization(llvm::Module&) + 1492
28 libLLVM.dylib            0x00000001084b8055 llvm::legacy::PassManagerImpl::run(llvm::Module&) + 981
29 opt                      0x0000000107fbafee main + 10327
30 libdyld.dylib            0x00007fff55d24115 start + 1
31 libdyld.dylib            0x000000000000000a start + 18446603339076321014
/Users/nika/Documents/REPOS/repo-predator-sym/sl/symheap.cc:3614: conditional breakpoint fired: CL_BREAK_IF(OBJ_RETURN != obj && !this->isValid(obj))
/Users/nika/Documents/REPOS/repo-predator-sym/sl/symheap.cc:3614: killing self by INT3 [SHA1 9961a7f2efe1549d2e476164f510df6c3844c1bc]
Trace/BPT trap: 5
```

> Let me create a separate pull request to get this fixed first.

Sure.

Comment (kdudka) [reply sl/symheap.hh:564]: You are right.  A debug build of Predator running on top of LLVM triggers a lot of assertion failures.  I have opened pull request #37 to improve the handling of anonymous stack objects.  A review would be appreciated!

**Comment by kdudka** [comment]
Merging, thanks!

**Pull request closed without merging**